### PR TITLE
Fix issue about checkerboardRasterCacheImages not working

### PR DESCRIPTION
**Pull request by ColdPaleLight**
fix https://github.com/flutter/flutter/issues/106946

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by flar — APPROVED** [review]
LGTM

**Comment by ColdPaleLight** [comment]
@flar I tweaked the patch to simplify the logic and added a unit test. Would you mind taking a look again? Thanks.

File: flow/layers/layer_raster_cache_item.cc [hunk 1]
```diff
@@ -152,7 +152,6 @@ bool LayerRasterCacheItem::TryToPrepareRasterCache(const PaintContext& context,
       .matrix             = matrix_,
       .logical_rect       = *paint_bounds,
       .flow_type          = flow_type,
-      .checkerboard       = context.checkerboard_offscreen_layers,
```
Comment (flar) [inline flow/layers/layer_raster_cache_item.cc:155]: Can we indent these lines the proper amount (4 spaces here).

**Comment by ColdPaleLight** [comment]
> Also, do we have a test that confirms the behavior with raster cache checkerboarding off, but offscreen checkerboarding on? That would be nice, especially since we just got the 2 confused.

Yes, those tests are here
https://github.com/flutter/engine/blob/main/flow/layers/checkerboard_layertree_unittests.cc

**Comment by flar** [comment]
> > Also, do we have a test that confirms the behavior with raster cache checkerboarding off, but offscreen checkerboarding on? That would be nice, especially since we just got the 2 confused.
> 
> Yes, those tests are here https://github.com/flutter/engine/blob/main/flow/layers/checkerboard_layertree_unittests.cc

I don't see them. All of those tests are outside of caching as far as I can tell...?

With offscreen vs rastercache checkerboarding being 2 flags we should be checking that all 4 combinations of

- rastercache checkerboarding of content that has checkerboarded offscreen layers
- layer-checkerboarding without any caching
- rastercache checkerboarding without any offscreen layers inside them
- neither

I think we already test the last 3, but we have no test of layer caching a layer that has a child that also has a saveLayer that will be offscreen-checkerboarded, do we?

**Review by flar — APPROVED** [review]
LGTM

**Pull request merged**